exporters/memory/include/opentelemetry/exporters/memory/in_memory_span_exporter.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "opentelemetry/sdk/trace/exporter.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace memory
+{
+/**
+ * A mock exporter that switches a flag once a valid recordable was received.
```

Comment:
nit: this is more of a fake then a mock
https://stackoverflow.com/questions/346372/whats-the-difference-between-faking-mocking-and-stubbing#:~:text=Fake%3A%20a%20class%20that%20implements,fixed%20data%20and%20no%20logic.&text=Mock%3A%20a%20class%20that%20implements,have%20been%20called%2Fnot%20called.

Reply:
Thanks - I've updated the terminology to reflect that this will be viewed as an in memory exporter as opposed to a mock.